olp-cpp-sdk-core/tests/thread/ContinuationTest.cpp

```diff
@@ -53,145 +56,167 @@ class ContinuationTest : public testing::Test {
-                          .Finally([&promise](Response<int> response) {
-                            promise.set_value(response.GetError());
+                          .Finally([&promise](ResponseType<int> response) {
+                            promise.set_value(std::move(response));
                           });
 
   continuation.CancelToken().Cancel();
   continuation.Run();
 
-  EXPECT_EQ(future.get().GetErrorCode(), olp::client::ErrorCode::Cancelled);
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  auto result = future.get();
+
+  ASSERT_EQ(result.GetError().GetErrorCode(),
+            olp::client::ErrorCode::Cancelled);
 }
 
 TEST_F(ContinuationTest, CancelAfterRun) {
-  std::promise<olp::client::ApiError> promise;
+  std::promise<ResponseType<int>> promise;
   auto future = promise.get_future();
 
   auto continuation = Create([](olp::thread::ExecutionContext,
                                 std::function<void(int)> next) { next(3); })
                           .Then([](olp::thread::ExecutionContext, int,
                                    std::function<void(int)> next) { next(3); })
-                          .Finally([&promise](Response<int> response) {
-                            promise.set_value(response.GetError());
+                          .Finally([&promise](ResponseType<int> response) {
+                            promise.set_value(response);
                           });
 
   continuation.Run();
+
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  auto result = future.get();
+
   continuation.CancelToken().Cancel();
 
-  EXPECT_EQ(future.get().GetErrorCode(), olp::client::ErrorCode::Cancelled);
+  EXPECT_TRUE(result);
 }
 
 TEST_F(ContinuationTest, CancelExecution) {
-  std::promise<void> promise;
+  std::promise<ResponseType<int>> promise;
   auto future = promise.get_future();
 
-  olp::client::ApiResponse<olp::client::HttpResponse, olp::client::ApiError>
-      result;
-
-  auto cont =
-      Create([](olp::thread::ExecutionContext, std::function<void(int)> next) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(200));
-        next(1);
-      })
-          .Finally([&promise, &result](
-                       olp::client::ApiResponse<int, olp::client::ApiError>
-                           response) {
-            result = response.GetError();
-            promise.set_value();
-          });
-
-  cont.Run();
-  std::this_thread::sleep_for(
-      std::chrono::milliseconds(50));  // simulate some delay
+  std::promise<void> cancel_promise;
+  auto cancel_future = cancel_promise.get_future();
+
+  auto continuation = Create([&cancel_future](olp::thread::ExecutionContext,
+                                              std::function<void(int)> next) {
+                        ASSERT_EQ(cancel_future.wait_for(kMaxWaitMs),
+                                  std::future_status::ready);
+                        cancel_future.get();
```

Comment:
You don't need the result, so `get()` call is useless.

Reply:
done